### PR TITLE
Feature/tiovx context singleton

### PR DESCRIPTION
**Pull request by michaelgruner**

### CODE REVIEW

File: gst-libs/gst/tiovx/gsttiovxcontext.c [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  gobject_class->finalize = gst_tiovx_context_finalize;
+}
+
+static GObject *
+gst_tiovx_context_constructor (GType type, guint n_construct_properties,
+    GObjectConstructParam * construct_properties)
+{
+  /* We cannot use the g_once trick here because it doesn't play
+     nicely with objects being destroyed entirely and created again,
+     which is a valid case of ourse. A normal mutex needs to be used.
```
Comment (emmadrigal) [inline gst-libs/gst/tiovx/gsttiovxcontext.c:110]: Typo `ours`

File: gst-libs/gst/tiovx/gsttiovxcontext.c [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+static void
+gst_tiovx_context_init (GstTIOVXContext * self)
+{
+  gint ret = 0;
+
+  GST_INFO ("Initializing TIOVX");
+
+  ret = appCommonInit ();
+  g_assert (ret == 0);
```
Comment (emmadrigal) [inline gst-libs/gst/tiovx/gsttiovxcontext.c:136]: `0 == ret`

File: gst-libs/gst/tiovx/gsttiovxcontext.c [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+static GObject *
+gst_tiovx_context_constructor (GType type, guint n_construct_properties,
+    GObjectConstructParam * construct_properties)
+{
+  /* We cannot use the g_once trick here because it doesn't play
+     nicely with objects being destroyed entirely and created again,
+     which is a valid case of ourse. A normal mutex needs to be used.
+   */
+  g_mutex_lock (&mutex);
```
Comment (emmadrigal) [inline gst-libs/gst/tiovx/gsttiovxcontext.c:112]: Don't we need to init this at some point?

**Review by emmadrigal — CHANGES_REQUESTED** [review]

**Review by rrcarlosrodriguez — APPROVED** [review]

**Review by emmadrigal — APPROVED** [review]

File: gst-libs/gst/tiovx/gsttiovxcontext.c (hunk 3 above)
Comment (michaelgruner) [reply gst-libs/gst/tiovx/gsttiovxcontext.c:112]: not when it is statically declared, according to [the docs](https://developer.gnome.org/glib/stable/glib-Threads.html#g-mutex-init):
> It is not necessary to initialize a mutex that has been statically allocated.

**Pull request closed without merging**